### ISSUE TITLE
mini conda 3 install with test on openSUSE 42.2 fails with 2 x "maximum recursion depth exceeded in comparison"

### ISSUE THREAD

**Issue by jetojedno** [conda/conda-build#2636]
### Actual Behavior

2 recipes fail with "maximum recursion depth exceeded in comparison"
Failed recipes:
  \- /home/david/Anaconda3/conda-bld/linux-64/libgcc-ng-7.2.0-h7cc24e2_2.tar.bz2
  \- /home/david/Anaconda3/conda-bld/linux-64/libstdcxx-ng-7.2.0-h7a57d05_2.tar.bz2

### Expected Behavior

All tests pass

### Steps to Reproduce

openSUSE 42.2 distribution with all updates applied

bash shell & KDE wm
rm -r ~/Anaconda3/* ; rm -r ~/Anaconda3/\\.*
./Miniconda3-latest-Linux-x86_64.sh -b -f -p ~/Anaconda3 -t

##### `Output of conda info`

```
     active environment : base
    active env location : /home/david/Anaconda3
            shell level : 1
       user config file : /home/david/.condarc
 populated config files : 
          conda version : 4.4.7
    conda-build version : 3.2.2
         python version : 3.6.3.final.0
       base environment : /home/david/Anaconda3  (writable)
           channel URLs : https://repo.continuum.io/pkgs/main/linux-64
                          https://repo.continuum.io/pkgs/main/noarch
                          https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
          package cache : /home/david/Anaconda3/pkgs
                          /home/david/.conda/pkgs
       envs directories : /home/david/Anaconda3/envs
                          /home/david/.conda/envs
               platform : linux-64
             user-agent : conda/4.4.7 requests/2.18.4 CPython/3.6.3 Linux/4.4.104-18.44-default opensuse/42.2 glibc/2.22
                UID:GID : 1000:100
             netrc file : None
           offline mode : False
```

**Comment by mingwandroid** [comment]
Do you have the first few lines of the backtrace available?

**Comment by jetojedno** [comment]
Full output 1:

`~> conda-build --override-channels --channel local --test /home/admin/Anaconda3/conda-bld/linux-64/libgcc-ng-7.2.0-h7cc24e2_2.tar.bz2
TEST START: /home/admin/Anaconda3/conda-bld/linux-64/libgcc-ng-7.2.0-h7cc24e2_2.tar.bz2
Updating index at /home/admin/Anaconda3/conda-bld/linux-64 to make package installable with dependencies
INFO:conda_build.build:Updating index at /home/admin/Anaconda3/conda-bld/linux-64 to make package installable with dependencies
Updating index at /home/admin/Anaconda3/conda-bld/noarch to make package installable with dependencies
INFO:conda_build.build:Updating index at /home/admin/Anaconda3/conda-bld/noarch to make package installable with dependencies
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /tmp/tmpfjcfipik/info/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /tmp/tmpfjcfipik/info/recipe/conda_build_config.yaml
Adding in variants from config.variant
INFO:conda_build.variants:Adding in variants from config.variant

Leaving build/test directories:
  Work:	 /home/admin/Anaconda3/conda-bld/work 
  Test:	 /home/admin/Anaconda3/conda-bld/test_tmp 
Leaving build/test environments:
  Test:	source activate  /home/admin/Anaconda3/conda-bld/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla 
  Build:	source activate  /home/admin/Anaconda3/conda-bld/_build_env 


maximum recursion depth exceeded in comparison`

**Comment by jetojedno** [comment]
Second:

`~> conda-build --override-channels --channel local --test /home/admin/Anaconda3/conda-bld/linux-64/libstdcxx-ng-7.2.0-h7a57d05_2.tar.bz2 
TEST START: /home/admin/Anaconda3/conda-bld/linux-64/libstdcxx-ng-7.2.0-h7a57d05_2.tar.bz2
Updating index at /home/admin/Anaconda3/conda-bld/linux-64 to make package installable with dependencies
INFO:conda_build.build:Updating index at /home/admin/Anaconda3/conda-bld/linux-64 to make package installable with dependencies
Updating index at /home/admin/Anaconda3/conda-bld/noarch to make package installable with dependencies
INFO:conda_build.build:Updating index at /home/admin/Anaconda3/conda-bld/noarch to make package installable with dependencies
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /tmp/tmp2uev8qg7/info/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /tmp/tmp2uev8qg7/info/recipe/conda_build_config.yaml
Adding in variants from config.variant
INFO:conda_build.variants:Adding in variants from config.variant

Leaving build/test directories:
  Work:	 /home/admin/Anaconda3/conda-bld/work 
  Test:	 /home/admin/Anaconda3/conda-bld/test_tmp 
Leaving build/test environments:
  Test:	source activate  /home/admin/Anaconda3/conda-bld/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla 
  Build:	source activate  /home/admin/Anaconda3/conda-bld/_build_env 


maximum recursion depth exceeded in comparison`

**Comment by jetojedno** [comment]
Also happening on openSUSE 42.3.

**Comment by msarahan** [comment]
I think I fixed this in conda-build 3.3.0.  Please update and try again.  I'm happy to reopen if this is still a problem.
